tests/common/snappi_tests/pfc_packet.py

```diff
@@ -0,0 +1,145 @@
+
+        Args:
+            pfc_frame_bytes (bytes): bytes of PFC frame
+        Returns:
+        """
+        cbfc_opcode = struct.unpack(">H", pfc_frame_bytes[0:2])[0]
+        class_enable_vec = struct.unpack(">H", pfc_frame_bytes[2:4])[0]
+        class_pause_times = []
+        for i in range(0, 16, 2):
+            class_pause_times.append(struct.unpack(">H", pfc_frame_bytes[i + 4:i + 6])[0])
+
+        self.cbfc_opcode = cbfc_opcode
+        self.class_enable_vec = _num_to_class_enable_vec_array(class_enable_vec)
+        self.class_pause_times = class_pause_times
+
+    def _check_cbfc_opcode(self):
+        """
+        Check if CBFC opcode is valid.
+
+        Args:
+
+        Returns:
+            True if valid CBFC opcode, False otherwise
+        """
+        if self.cbfc_opcode == 0x0101:
+            return True
+        else:
+            return False
+
+    def _check_class_enable_vec(self):
+        """
+        Check if class enable vector is valid i.e. either each bit is 0 or 1.
+
+        Args:
+            class_enable_vec (list of chars): class enable vector
+
+        Returns:
+            True if valid class enable vector, False otherwise
+        """
+        valid_options = ["0", "1"]
+        for val in self.class_enable_vec:
+            if val not in valid_options:
+                return False
+
+        return False
+
+    def _check_class_pause_times(self):
+        """
+        Check if class pause times are valid. Both conditions must be met:
+        1) class pause times are between 0x0 and 0xFFFF
+        2) class pause times are 0 if the corresponding bit in the class enable vector is 0, and vice versa
+
+        Args:
+
+        Returns:
+            True if valid class pause times, False otherwise
+        """
+        for i in range(len(self.class_pause_times)):
+            if self.class_pause_times[i] < 0x0 and self.class_pause_times[i] > 0xFFFF:
+                return False
+            elif self.class_pause_times[i] > 0x0 and self.class_enable_vec[PRIO_DEFAULT_LEN - i - 1] == "0":
+                return False
+            elif self.class_pause_times[i] == 0x0 and self.class_enable_vec[PRIO_DEFAULT_LEN - i - 1] == "1":
+                return False
+
+        return True
+
+    def validate_pfc_frame(self):
+        """
+        Validate the PFC frame. The PFC frame is valid if:
+        1) CBFC opcode is 0x0101
+        2) class enable vector is valid
+        3) class pause times are valid
+
+        Check function subdefinitions for more details.
+        """
+        is_valid_cbfc_opcode = self._check_cbfc_opcode()
+        is_valid_class_enable_vec = self._check_class_enable_vec()
+        is_valid_class_pause_times = self._check_class_pause_times()
+        if not is_valid_cbfc_opcode or not is_valid_class_enable_vec or not is_valid_class_pause_times:
+            self.is_valid_frame = False
+        else:
+            self.is_valid_frame = True
+
```

Comment:
It seems that we should add a function called is_valid to return is_valid_frame.

Reply:
done